### PR TITLE
Port Trigger Editor from Elm to React

### PR DESCRIPTION
**Pull request by davidebriani**
This PR adds a Javascript / React implementation to port TriggerEditor pages from Elm to React.
To do so, it:
- adds an AstarteTrigger model in `astarte-client`, together with validation and transforms for API communication
- adds a reusable TriggerEditor React component to create and display triggers
- uses TriggerEditor component to implement the New Trigger page in React
- uses TriggerEditor component to implement the Edit Trigger page in React
- removes unused Elm files from the codebase

### CODE REVIEW

File: src/astarte-client/client.ts [hunk 1]
```diff
@@ -254,6 +262,13 @@ class AstarteClient {
     }
   }
 
+  getCredentials(): { realm: string; token: string } {
```
[the remark this answers is not in the document]
Comment (davidebriani) [reply src/astarte-client/client.ts]: Reworked and removed `getCredentials()`. Now a `realm(): string | null` getter is exposed.

File: src/react/components/TriggerEditor/index.tsx [hunk 2]
```diff
@@ -0,0 +1,1479 @@
...
+
+  useEffect(() => {
+    if (initialData) {
+      handleFetchInterfacesForTrigger(initialData);
+    } else {
+      handleFetchInterfacesName();
+    }
+  }, []);
+
+  return (
```
[the remark this answers is not in the document]
Comment (davidebriani) [reply src/react/components/TriggerEditor/index.tsx:430]: Reworked and extracted two sub-components: `<TriggerActionForm />` and `<SimpleTriggerForm />`

File: src/react/components/TriggerEditor/index.tsx [hunk 3]
```diff
@@ -0,0 +1,1479 @@
...
+  const [triggerInterface, setTriggerInterface] = useState<AstarteInterface | null>(null);
+  const [isLoadingInterfacesName, setIsLoadingInterfacesName] = useState(false);
+  const [isLoadingInterfaceMajors, setIsLoadingInterfaceMajors] = useState(false);
+  const [isLoadingInterface, setIsLoadingInterface] = useState(false);
+  const [isNewAmqpHeaderModalVisible, setIsNewAmqpHeaderModalVisible] = useState(false);
+  const [isNewHttpHeaderModalVisible, setIsNewHttpHeaderModalVisible] = useState(false);
+  const [actionAmqpHeaderToEdit, setActionAmqpHeaderToEdit] = useState<string | null>(null);
+  const [actionHttpHeaderToEdit, setActionHttpHeaderToEdit] = useState<string | null>(null);
+  const [actionAmqpHeaderToRemove, setActionAmqpHeaderToRemove] = useState<string | null>(null);
+  const [actionHttpHeaderToRemove, setActionHttpHeaderToRemove] = useState<string | null>(null);
```
[the remark this answers is not in the document]
Comment (davidebriani) [reply src/react/components/TriggerEditor/index.tsx]: Reworked so we now have just one state to track, and just one modal open at a time.

File: src/react/components/TriggerEditor/index.tsx [hunk 4]
```diff
@@ -0,0 +1,545 @@
...
+                  <Form.Control.Feedback type="invalid">
+                    {_.get(triggerValidationErrors, 'name')}
+                  </Form.Control.Feedback>
+                </Form.Group>
+              </Col>
+            </Form.Row>
+          </Form>
+          <SimpleTriggerForm
+            fetchInterface={handleFetchInterface}
+            fetchInterfacesName={handleFetchInterfacesName}
```
Comment (Pavinati) [inline src/react/components/TriggerEditor/index.tsx:457]: this is only done once as soon as you open the page. You can fetch the list of interfaces in the main component, and pass it down as a property.

Comment (davidebriani) [reply src/react/components/TriggerEditor/index.tsx]: Nice catch, I fixed it. Now the list of interfaces is passed down as a property, no re-fetching needed.

File: src/react/components/TriggerEditor/index.tsx [hunk 5]
```diff
@@ -0,0 +1,544 @@
...
+  const handleTriggerNameChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = e.target;
+    setTriggerDraft((draft) => ({ ...draft, name: value }));
+  }, []);
+
+  const handleAddActionAmqpHeader = useCallback(() => {
+    setActiveModal({ modal: 'new-amqp-header' });
+  }, []);
+
+  const handleDismissNewAmqpHeaderModal = useCallback(() => {
```
Comment (Pavinati) [inline src/react/components/TriggerEditor/index.tsx:276]: handleDismissNewAmqpHeaderModal, handleDismissEditAmqpHeaderModal and handleDismissNewHttpHeaderModal all act the same.
If we just need to dismiss the modal, we could define a `dismissModal` and reuse it.

Comment (davidebriani) [reply src/react/components/TriggerEditor/index.tsx]: Fixed it.

**Review by Pavinati — APPROVED** [review]

**Review by bettio — APPROVED** [review]

**Pull request merged**